archimedes/archimedes/functions/plotting/scatter_plot.py

```diff
@@ -0,0 +1,66 @@
+import plotly.express as px
+
+from .colors import colors
+from ..utils import do_call
+from ..list import unique, order
+
+def scatter_plotly(
+    data_frame, x_by: str, y_by: str, color_by: str,
+    px_args: dict = {},
+    size = 5, color_panel: list = colors,
+    color_order_legend: str = 'increasing',
+    plot_title: str = None, legend_title: str = None,
+    xlab: str = None, ylab: str = None
+    ):
+    """
+    Produces a scatter plot using plotly.express.scatter based on the pandas 'data_frame' given.
+    'x_by', 'y_by', and 'color_by' should indicate columns of 'data_frame' to use for x/y/color.
+    'px_args' should be a dictionary of additional bits to send in the 'plotly.express.scatter' call.
+    'size' sets the size of points.  Can be either a number directly or the name of a column of 'data_frame'.
+    'color_panel' (string list) sets the colors when 'color_by' references discrete data
+    'color_order_legend' ('increasing', 'decreasing', or 'unordered') sets the ordering of keys in the legend, when 'color_by' references discrete data
+    'plot_title', 'legend_title', 'xlab', and 'ylab' set titles.
+    """
+
+    # Parse dependent defaults (given 'None' b/c python)
+    def fill_by_if_None_and_logic(this, fill, logic = True):
+        if (this == None and logic):
+            return fill
+        return this
+    xlab = fill_by_if_None_and_logic(xlab, x_by)
+    ylab = fill_by_if_None_and_logic(ylab, y_by)
+    plot_title = fill_by_if_None_and_logic(plot_title, color_by)
+    legend_title = fill_by_if_None_and_logic(legend_title, color_by)
+
+    # Add to px_args (Can probably remove this section once I learn python syntax better!)
+    px_args['data_frame'] = data_frame
+    px_args['x'] = x_by
+    px_args['y'] = y_by
+    px_args['color'] = color_by
+    px_args['color_discrete_sequence'] = color_panel
+
+    # FUTURE! Make this work for bool too.  And maybe other types
+    if not all(map(lambda x: isinstance(x, str), data_frame[color_by])):
+        color_order_legend = 'unordered'
+    if (color_order_legend == 'increasing'):
```

Comment:
Question posed above is addressed here.  Not sure if there might be a better way, but the ordering code that follows only works for strings so I have it ensuring the data is string-type